### PR TITLE
SDKUpdater Improvements

### LINKED ISSUE #666: SDK Updater crahes with 'java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema' error
Due to removal of Java EE modules in Java 11:

https://jaxenter.com/jdk-11-java-ee-modules-140674.html

This could be fixed by using new new Android SDK Command-line Tools For JVM 9, 10, 11, And Later:

https://www.dev2qa.com/how-to-fix-java-lang-noclassdeffounderror-javax-xml-bind-annotation-xmlschema-when-install-android-sdk/

### PR DESCRIPTION
**Pull request by rupeshkumar22**
Fixes #666, #718  

The PR introduces the following changes-

- Progress Bar has been set to determine mode with percentage progress
- A new secondary status has been added for process details
- UI has been improved(Action Button was not showing the full text)
- Size reduction: Only the necessary jar files have been included in AndroidMode's final zip. Also, runtime testing has been done to make sure only necessary jars are included rest are excluded with exclude statements in `SDKUpdater/build.xml` Including all the jars costs 50+ MB  in total for AndroidMode.zip which has been reduced to approx 32 MB.
- Removed java modules in java 17 have been supported with necessary jars.(#666)
- [android-toolslib](https://github.com/processing/processing-android/blob/b807ab8e71fded580286b20df411c07ed90e7cb2/mode/version.properties#L3) has been updated to 30.3.0 from 27.1.1 and corresponding methods (i.e AndroidSDKHandler) have been updated accordingly.
- Now a `CustomProgressIndicatorToMonitor` class is being used and with the help of its overridden methods, we are able to reflect the progress on UI.

SDKUpdaterPR

An outcome of Google Summer of Code 2022
Author - Rupesh Kumar @rupeshkumar22
Mentor - Aditya Rana @ranaaditya
Project Lead(Processing For Android) - Andrés Colubri @codeanticode

### CODE REVIEW

File: mode/tools/SDKUpdater/build.gradle [hunk 1]
```diff
@@ -12,21 +12,57 @@ dependencies {
...
-    include '**/dvlib-*'
-    include '**/gson-*'
-    include '**/guava-*'
-    include '**/httpcore-*'
-    include '**/kotlin-stdlib-*'
-    include '**/layoutlib-api-*'
-    include '**/repository-*'
-    include '**/sdklib-*'
+        include '**/*.jar'
+        exclude '**/annotations-*'
```
Comment (ranaaditya) [inline mode/tools/SDKUpdater/build.gradle:22]: just take care of wat you are removing from the final fat jar. And to test it you can try android-mode as external library in an android project.

Make sure that no dependency get excluded from the final fat jar.

Thanks !

**Review by ranaaditya — APPROVED** [review]

File: mode/tools/SDKUpdater/build.gradle (hunk 1 above)
Comment (rupeshkumar22) [reply mode/tools/SDKUpdater/build.gradle:22]: sure Aditya, I started by removing the jars, and whenever I got the class not found the issue, I added those specific class jars. I made sure both the Query and Download Swing Worker tasks went well. But I will do some more testing as you suggested to make sure no necessary dependency gets excluded.

Thanks for highlighting the importance

Comment (ranaaditya) [reply mode/tools/SDKUpdater/build.gradle:22]: > sure Aditya, I started by removing the jars, and whenever I got the class not found the issue, I added those specific class jars. I made sure both the Query and Download Swing Worker tasks went well. But I will do some more testing as you suggested to make sure no necessary dependency gets excluded.
> 
> Thanks for highlighting the importance

hey @rupeshkumar22 any update on this ?

Thanks !

Comment (rupeshkumar22) [reply mode/tools/SDKUpdater/build.gradle:22]: Hi @ranaaditya 
I am done with the testing and the updater faces no issues in fetching/installing the SDK data and reflecting back on the updater's UI. The included libraries in build.gradle are the only necessary and sufficient libraries to do the required operation of fetch or install.

**Comment by rupeshkumar22** [comment]
Better to use only `include *.jar` statements. So adding one more commit for SDKUpdater.

**Comment by rupeshkumar22** [comment]
I'm resolving Issue #718 as a part of this PR So that there won't be any conflicts.

**Comment by rupeshkumar22** [comment]
Testing has been done from my side after applying the new changes. Also checked the updater again. Tried installing SDK with the updater. No issue has been faced. 
Also, Fixed #718

**Comment by codeanticode** [comment]
@rupeshkumar22 Thank you for the PR, I'm merging into a separate branch for testing before making the new release. Happy New Year :-)

**Comment by rupeshkumar22** [comment]
@codeanticode Happy New Year! thanks :)

**Pull request merged**